openmdao/recorders/base_recorder.py

```diff
@@ -387,6 +392,71 @@ def record_iteration(self, object_requesting_recording, metadata, **kwargs):
         else:
             raise ValueError("Recorders must be attached to Drivers, Systems, or Solvers.")
 
+    def record_iteration_driver_passing_vars(self, object_requesting_recording, desvars, responses,
+                                             objectives, constraints, metadata):
+        """
+        Record an iteration using the Driver options.
+
+        Parameters
+        ----------
+        object_requesting_recording : object
+            The Driver in need of recording.
+        metadata : dict, optional
+            Dictionary containing execution metadata.
+        desvars: dict
+            The design variables of the Driver being recorded.
+        responses: dict
+            The responses of the Driver being recorded.
+        objectives: dict
+            The objectives of the Driver being recorded.
+        constraints: dict
+            The constraints of the Driver being recorded.
+        """
+        # TODO: this code and the same code in record_iteration should be in a separate method
+        if not self._parallel:
+            if MPI and MPI.COMM_WORLD.rank > 0:
+                raise RuntimeError("Non-parallel recorders should not be recording on ranks > 0")
+
+        self._counter += 1
+        self._iteration_coordinate = get_formatted_iteration_coordinate()
+
+        if self.options['record_desvars']:
+            if self._filtered_driver:
+                self._desvars_values = \
+                    {name: desvars[name] for name in self._filtered_driver['des']}
+            else:
+                self._desvars_values = desvars
+        else:
+            self._desvars_values = None
+
+        # Cannot handle responses yet
+        # if self.options['record_responses']:
+        #     if self._filtered_driver:
+        #         self._responses_values = \
+        #             {name: responses[name] for name in self._filtered_driver['res']}
+        #     else:
+        #         self._responses_values = responses
+        # else:
+        #     self._responses_values = None
+
+        if self.options['record_objectives']:
+            if self._filtered_driver:
+                self._objectives_values = \
+                    {name: objectives[name] for name in self._filtered_driver['obj']}
+            else:
+                self._objectives_values = objectives
+        else:
+            self._objectives_values = None
+
+        if self.options['record_constraints']:
+            if self._filtered_driver:
+                self._constraints_values = \
+                    {name: constraints[name] for name in self._filtered_driver['con']}
+            else:
+                self._constraints_values = constraints
+        else:
+            self._constraints_values = None
+
     def record_iteration_driver(self, object_requesting_recording, metadata):
```

Comment:
Currently it looks like there is a `record_iteration_driver` and a `record_iteration_driver_passing_vars` in base_recorder... the "routing" function `record_iteration` still has a call to `record_iteration_driver` in it.  But the two functions appear to be very similar.  Should the new one just replace the old one, and also just assume its old name?